Mathlib/FieldTheory/Tower.lean

```diff
@@ -119,19 +73,12 @@ theorem right [hf : FiniteDimensional F A] : FiniteDimensional K A :=
 #align finite_dimensional.right FiniteDimensional.right
 
 /-- Tower law: if `A` is a `K`-vector space and `K` is a field extension of `F` then
-`dim_F(A) = dim_F(K) * dim_K(A)`.
-
-This is `FiniteDimensional.finrank_mul_finrank'` with one fewer finiteness assumption. -/
-theorem finrank_mul_finrank [FiniteDimensional F K] : finrank F K * finrank K A = finrank F A := by
-  by_cases hA : FiniteDimensional K A
-  · replace hA : FiniteDimensional K A := hA -- porting note: broken instance cache
-    rw [finrank_mul_finrank']
-  · rw [finrank_of_infinite_dimensional hA, mul_zero, finrank_of_infinite_dimensional]
-    exact mt (@right F K A _ _ _ _ _ _ _) hA
+`dim_F(A) = dim_F(K) * dim_K(A)`. -/
```

Comment:
```suggestion
`dim_F(A) = dim_F(K) * dim_K(A)`. 
  A special case of `FiniteDimensional.finrank_mul_finrank'` which assumes `Module.Free`.
We keep this because Lean sometimes has trouble finding the `Module.Free` instances. -/
```